### PR TITLE
Make Deno.Buffer implement only sync I/O

### PR DESCRIPTION
**Pull request by jcao219**
Deno.Buffer shouldn't implement async read and write

https://github.com/denoland/deno/issues/4933

### CODE REVIEW

**Comment by jcao219** [comment]
Tests can be fixed after CI run

**Comment by jcao219** [comment]
So I went ahead and removed Deno.readFrom -- I can add it back in if that makes sense -- but from reading the source code it seems that it doesn't make sense to me.

**Comment by jcao219** [comment]
image

Well, looks like I'll need to open a command line after all...

**Comment by jcao219** [comment]
Tests can be dealt with tomorrow

File: cli/js/buffer.ts [hunk 1]
```diff
@@ -161,25 +151,6 @@ export class Buffer implements Reader, SyncReader, Writer, SyncWriter {
...
-          return n;
-        }
-        this.#reslice(i + nread);
-        n += nread;
-      } catch (e) {
-        return n;
-      }
-    }
-  }
-
```
Comment (bartlomieju) [inline cli/js/buffer.ts:182]: This method should stay

File: cli/js/buffer.ts [hunk 2]
```diff
@@ -200,12 +171,6 @@ export class Buffer implements Reader, SyncReader, Writer, SyncWriter {
   }
 }
 
-export async function readAll(r: Reader): Promise<Uint8Array> {
-  const buf = new Buffer();
-  await buf.readFrom(r);
-  return buf.bytes();
-}
-
```
Comment (bartlomieju) [inline cli/js/buffer.ts:208]: Ditto

File: cli/js/net.ts [hunk 3]
```diff
@@ -44,6 +44,11 @@ export class ConnImpl implements Conn {
     return read(this.rid, p);
   }
 
+  readSync(p: Uint8Array): number | EOF {
+    // TODO(jcao219): Verify that readSync op works for all conn rids.
```
Comment (jcao219) [inline cli/js/net.ts:48]: I looked at the rust code and it seems clear that it doesn't

File: cli/js/web/fetch.ts [hunk 4]
```diff
@@ -224,6 +230,11 @@ class Body implements domTypes.Body, ReadableStream<Uint8Array>, io.ReadCloser {
     return read(this.#rid, p);
   }
 
+  readSync(p: Uint8Array): number | io.EOF {
```
Comment (jcao219) [inline cli/js/web/fetch.ts:233]: revert this

File: cli/js/buffer.ts [hunk 5]
```diff
@@ -161,25 +151,6 @@ export class Buffer implements Reader, SyncReader, Writer, SyncWriter {
...
   }
 
-  async readFrom(r: Reader): Promise<number> {
-    let n = 0;
-    while (true) {
-      try {
-        const i = this.#grow(MIN_READ);
-        this.#reslice(i);
-        const fub = new Uint8Array(this.#buf.buffer, i);
-        const nread = await r.read(fub);
```
Comment (ry) [inline cli/js/buffer.ts:171]: Sorry, I wasn't very specific, readFrom should still be async. Just Buffer should remove its read() and write() methods.

**Comment by jcao219** [comment]
Thanks ryan and bart, tonight i'll continue

**Comment by ry** [comment]
@jcao219 Thanks - please hold off on continuing this one. We've had some discussions regarding buffer and we're not sure if we want to remove these methods anymore. Sorry for the confusion - I'll update later.

**Comment by jcao219** [comment]
No worries! I'm glad I got to take another look at the codebase and see how
much it's improved.

**Comment by bartlomieju** [comment]
@jcao219 thanks once again for the patch! We've decided to keep async interfaces for `Deno.copy()` compatibility.

**Comment by jcao219** [comment]
You bet. Good call!

**Pull request closed without merging**